libbeat/docs/monitoring/configuring.asciidoc

```diff
@@ -0,0 +1,178 @@
+If set to `true`, monitoring is enabled.
+
+The default value is `false`.
+
+[float]
+==== `elasticsearch`
+
+The {es} instances that you want to ship your {beatname_uc} metrics to. This
+configuration option contains the following fields:
+
+[float]
+===== `bulk_max_size`
+
+The maximum number of metrics to bulk in a single {es} bulk API index request.
+The default is `50`. For more information, see <<elasticsearch-output>>.
+
+////
+[float]
+===== `buffer_size`
+
+TBD. The default value is `50`.
+////
+[float]
+===== `compression_level`
+
+The gzip compression level. Setting this value to `0` disables compression. The
+compression level must be in the range of `1` (best speed) to `9` (best
+compression). The default value is `0`. Increasing the compression level
+reduces the network usage but increases the CPU usage.
+
+[float]
+===== `headers`
+
+Custom HTTP headers to add to each request. For more information, see
+<<elasticsearch-output>>.
+
+[float]
+===== `hosts`
+
+The list of {es} nodes to connect to. Monitoring metrics are distributed to
+these nodes in round robin order. For more information, see
+<<elasticsearch-output>>.
+
+[float]
+===== `max_retries`
+
+The number of times to retry sending the monitoring metrics after a failure.
+After the specified number of retries, the metrics are typically dropped. The
+default value is `3`. For more information, see <<elasticsearch-output>>.
+
+[float]
+===== `parameters`
+
+Dictionary of HTTP parameters to pass within the url with index operations.
+
+[float]
+===== `password`
+
+The password that {beatname_uc} uses to authenticate with the {es} instances for
+shipping monitoring data.
+////
+[float]
+===== `period`
+
+TBD. The default value is 60 * time.Second.
+////
+[float]
+===== `protocol`
+
+The name of the protocol to use when connecting to the {es} cluster. The options
+are: `http` or `https`. The default is `http`. If you specify a URL for `hosts`,
+however, the value of protocol is overridden by the scheme you specify in the URL.
+
+[float]
+===== `proxy_url`
+
+The URL of the proxy to use when connecting to the {es} cluster. For more
+information, see <<elasticsearch-output>>.
+////
+[float]
+===== `tags`
+
+TBD
+////
+
+[float]
+===== `timeout`
+
+The HTTP request timeout in seconds for the {es} request. The default is `90`.
+
+[float]
+===== `ssl`
+
+Configuration options for Transport Layer Security (TLS) or Secure Sockets Layer
+(SSL) parameters like the certificate authority (CA) to use for HTTPS-based
+connections. If the `ssl` section is missing, the host CAs are used for
+HTTPS connections to {es}. For more information, see <<configuration-ssl>>.
+
+[float]
+===== `url`
+
+The URL for the {es} instances. You can specify a single host as a string or
+specify multiple hosts as an array. Defaults to `http://localhost:9200`.
+
+[float]
+===== `username`
+
+The user ID that {beatname_uc} uses to authenticate with the {es} instances for
+shipping monitoring data.
```

Comment:
Note: this is by default `beats_system`. X-Pack automatically creates a `beats-system` user for use with monitoring. The password should be configured, as X-Pack creates a random password at install time.